src/commands/fal.rs

```diff
@@ -0,0 +1,215 @@
+            log::error!("fal.ai error: {:#?}", e);
+            e
+        })?;
+        let image = process_image(&generation);
+        let mut temp_file = NamedTempFile::new().unwrap();
+        image.write_to(&mut BufWriter::new(&mut temp_file), ImageFormat::Jpeg).unwrap();
+
+        let message = ctx
+            .reply_custom(
+                InputMessageContent::InputMessagePhoto(InputMessagePhoto {
+                    photo: InputFile::Local(InputFileLocal {
+                        path: temp_file.path().to_str().unwrap().into(),
+                    }),
+                    thumbnail: None,
+                    added_sticker_file_ids: Vec::new(),
+                    width: image.width().try_into().unwrap(),
+                    height: image.height().try_into().unwrap(),
+                    caption: Some(format_result_text(generation)),
+                    show_caption_above_media: false,
+                    self_destruct_type: None,
+                    has_spoiler: false,
+                }),
+                Some(ReplyMarkup::InlineKeyboard(ReplyMarkupInlineKeyboard {
+                    rows: vec![vec![InlineKeyboardButton {
+                        text: String::from("generated by fal.ai"),
+                        r#type: InlineKeyboardButtonType::Url(InlineKeyboardButtonTypeUrl {
+                            url: "https://fal.ai/".into(),
+                        }),
+                    }]],
+                })),
+            )
+            .await?;
+
+        ctx.bot_state.message_queue.wait_for_message(message.id).await?;
+        temp_file.close().unwrap();
+
+        Ok(())
+    }
+}
+
+impl Fal {
+    async fn generate(
+        &self,
+        ctx: &CommandContext,
+        prompt: String,
+    ) -> Result<Generation, CommandError> {
+        let request = FalRequest {
+            model_name: self.model_name,
+            prompt,
+            negative_prompt: String::new(),
+            image_size: ImageSize {
+                height: self.size.0,
+                width: self.size.1,
+            },
+            num_inference_steps: self.num_inference_steps,
+            guidance_scale: 5,
+            loras: vec![],
+            embeddings: vec![],
+            num_images: 1, // this variable is potentially changable - this is so cheap anyways
+            enable_safety_checker: true,
+            format: "jpeg",
+            safety_checker_version: "v1",
+        };
+
+        let response = ctx
+            .bot_state
+            .http_client
+            .post(format!("https://fal.run/fal-ai/{}", self.model_name))
+            .header("Authorization", format!("Key {}", env::var("FAL_API_KEY").unwrap()))
+            .json(&request)
+            .send()
+            .await?
+            .server_error()?;
+
+        if response.status() == StatusCode::OK {
+            let response = response.json::<FalResponse>().await?;
+            Ok(Generation {
+                image: response.images[0].clone(),
```

Comment:
don't clone the data
```suggestion
                image: response.images.into_iter().next().unwrap(),
```